### PR TITLE
Output BankHashDetails file when leader drops its' own block

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
    Currently, the file is generated when a node drops a block that was
    produced by another node. However, it would also be beneficial to see
    the account state when a node drops its' own block.

#### Summary of Changes
    Output the file in this additional failure codepath
    
Note that the debug file gets generated when we are not the leader via `ReplayStage::purge_unconfirmed_duplicate_slot()`:
https://github.com/solana-labs/solana/blob/e425c1acaadf6ac22aec3afa09750616443349f1/core/src/replay_stage.rs#L1510

### CODE REVIEW

File: core/src/replay_stage.rs [hunk 1]
```diff
@@ -1349,14 +1349,18 @@ impl ReplayStage {
...
-
                     // Should not dump slots for which we were the leader
                     if Some(*my_pubkey) == leader_schedule_cache.slot_leader_at(*duplicate_slot, None) {
-                            panic!("We are attempting to dump a block that we produced. \
-                                This indicates that we are producing duplicate blocks, \
-                                or that there is a bug in our runtime/replay code which \
-                                causes us to compute different bank hashes than the rest of the cluster. \
-                                We froze slot {duplicate_slot} with hash {frozen_hash:?} while the cluster hash is {correct_hash}");
+                        if let Some(duplicate_bank) = bank_forks.read().unwrap().get(*duplicate_slot) {
+                            let _ = bank_hash_details::write_bank_hash_details_file(&duplicate_bank);
```
Comment (AshwinSekar) [inline core/src/replay_stage.rs:1355]: nit: instead of ignoring the `Result` should we log the details of why we can't write the bank hash details?

Comment (steviez) [reply core/src/replay_stage.rs]: Good call - done in https://github.com/solana-labs/solana/pull/34256/commits/4ef6b525f88715be19f71a2cf7c4e48050adb419

**Review by AshwinSekar — APPROVED** [review]
LGTM. if we're backporting the regular bank hash details, I vote that we should also backport this pr as it is extremely useful in debugging.

**Comment by steviez** [comment]
> LGTM. if we're backporting the regular bank hash details, I vote that we should also backport this pr as it is extremely useful in debugging.

Agreed, that is currently a work in progress 😉

**Pull request merged**